azkaban-common/src/test/java/azkaban/database/AzkabanConnectionPoolTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.database;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import org.apache.commons.dbutils.DbUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+
+public class AzkabanConnectionPoolTest{
+
+  public static class EmbeddedDerbyBasicDataSource extends AzkabanDataSource {
+
+    private EmbeddedDerbyBasicDataSource() {
+      super();
+      String url = "jdbc:derby:memory:MyDatabase;create=true";
+      setDriverClassName("org.apache.derby.jdbc.EmbeddedDriver");
+      setUrl(url);
+    }
+
+    @Override
+    public boolean allowsOnDuplicateKey() {
+      return false;
+    }
+
+    @Override
+    public String getDBType() {
+      return "Derby";
+    }
+  }
+
+  AzkabanDataSource derbyDataSource;
+
+  @Before
+  public void setup() throws Exception{
+    derbyDataSource = new EmbeddedDerbyBasicDataSource();
+  }
+
+  @Test
+  public void testCoonectionPoolAutoCommitRest() throws Exception {
+    Connection connection = derbyDataSource.getConnection();
+    Assert.assertEquals(connection.getAutoCommit(), true);
```

Comment:
This works.

It would be even better if each test stays focused and test only one thing if possible so that it is easier to read and debug.